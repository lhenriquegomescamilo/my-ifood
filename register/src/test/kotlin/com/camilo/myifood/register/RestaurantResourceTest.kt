package com.camilo.myifood.register

import com.github.database.rider.cdi.api.DBRider
import com.github.database.rider.core.api.configuration.DBUnit
import com.github.database.rider.core.api.configuration.Orthography
import com.github.database.rider.core.api.dataset.DataSet
import io.quarkus.test.common.QuarkusTestResource
import io.quarkus.test.junit.QuarkusTest
import io.restassured.RestAssured.given
import org.apache.http.HttpStatus
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(RegisterTestLifecycleManager::class)
class RestaurantResourceTest {

    @Test
    @DataSet(value = ["restaurants.yml"])
    fun `should return the list of restaurants`() {
        val result = given()
            .`when`().get("/restaurants")
            .then()
            .statusCode(HttpStatus.SC_OK)
            .extract().asString()
        Approvals.verifyJson(result)
    }



}