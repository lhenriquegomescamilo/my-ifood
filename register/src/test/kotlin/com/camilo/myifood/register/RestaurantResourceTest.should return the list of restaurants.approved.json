[
  {
    "cnpj": "10.335.524/0001-97",
    "foods": [],
    "id": 1,
    "name": "New Restaurant",
    "owner": "id keyclock "
  },
  {
    "cnpj": "93.760.087/0001-86",
    "foods": [],
    "id": 2,
    "name": "New Restaurant 2",
    "owner": "id keyclock 2 "
  }
]